src/main/groovy/edu/oregonstate/mist/slashcourses/resources/SlashCourseResource.groovy

```diff
@@ -134,4 +142,51 @@ class SlashCourseResource extends Resource {
         }
         returnResponse
     }
+
+    /**
+     * Respond to PUT requests by updating a course object according to course CRN.
+     * If the course object doesn't exist, create one with POST requests.
+     *
+     * @param crn
+     * @param newCourse
+     * @return response containing the result or error message
+     */
+    @PUT
+    @Path('{crn: \\d+}')
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response putByCRN (@PathParam('crn') Integer crn, SlashCourse newCourse) {
```

Comment:
I think you can use the @Valid notation to make sure that the user provides a valid SlashCourse. Is there a reason why this method can't use it?